feature/session/src/main/java/io/github/droidkaigi/confsched2019/session/ui/widget/TimetableLayoutManager.kt

```diff
@@ -227,6 +283,48 @@ class TimeTableLayoutManager(
         return dx
     }
 
+    private fun layoutChildren(recycler: Recycler) {
+        val offsetY = parentTop
+        var offsetX = parentLeft
+        for (i in 0 until columns.size()) {
+            val columnNumber = columns.keyAt(i)
+            if (i == 0) anchor.leftColumn = columnNumber
+            offsetX += fillColumnHorizontally(columnNumber, 0, offsetX, offsetY, true, recycler)
+            if (offsetX > parentRight) {
+                anchor.rightColumn = columnNumber
+                break
+            }
+        }
+    }
+
+    private fun relayoutChildren(startX: Int, startY: Int, topPeriod: Period, recycler: Recycler) {
+        var offsetX = startX
+        anchor.leftColumn = topPeriod.columnNumber
+        val columnCount = columns.size()
+        var i = columns.indexOfKey(topPeriod.columnNumber)
+        while (i < columnCount) {
+            val columnNumber = columns.keyAt(i)
+            val startPositionInColumn =
+                calculateStartPeriodInColumn(columnNumber, startY, topPeriod) ?: continue
+            val offsetY = startY +
+                (startPositionInColumn.startUnixMin - topPeriod.startUnixMin) * pxPerMinute
+            offsetX += fillColumnHorizontally(
+                columnNumber,
+                startPositionInColumn.positionInColumn,
+                offsetX,
+                offsetY,
+                true,
+                recycler
+            )
+
+            if (i == columnCount - 1) i = 0 else i++
```

Comment:
I'm not sure whether or not there is a huge device though, the following sounds better. What do you think ?

```
...
var iterateCount = 0
while (... && !shouldLoopHorizontally && iterateCount < columnCount) {
  ...
  iterateCount++
}
```

Reply:
after thinking about it, it will not work on large screen devices.
maybe, columns will be drawn like this...
`|3|4|5|6|7|8|0|1|2|3|4|`

I will rewrite.